### PR TITLE
126 Update API Route Security to use Annotations and Proper Roles

### PR DESCRIPTION
**Pull request by uplyft**
First attempt at assigning the appropriate roles, but in some cases I had to guess. Interested to see if this is even close to what we need.

### CODE REVIEW

File: src/Controller/PartnerController.php [hunk 1]
```diff
@@ -122,6 +135,8 @@ public function update(Request $request, int $id)
      * Delete a Partner
      *
      * @Route(path="/{id}", methods={"DELETE"})
+     * @IsGranted({"ROLE_PARTNER_EDIT_ALL","ROLE_PARTNER_MANAGE_OWN"})
```
Comment (akoebbe) [inline src/Controller/PartnerController.php:138]: Hmm... Not sure we want Partners deleting themselves. Let's pull `ROLE_PARTNER_MANAGE_OWN` off.

File: src/Controller/PartnerController.php [hunk 2]
```diff
@@ -37,7 +41,10 @@ public function index(Request $request)
     }
 
     /**
+     *
      * @Route("", methods={"POST"})
+     * @IsGranted({"ROLE_PARTNER_EDIT_ALL","ROLE_PARTNER_MANAGE_OWN"})
```
Comment (akoebbe) [inline src/Controller/PartnerController.php:46]: I think we can pull off `ROLE_PARTNER_MANAGE_OWN` on this. We will eventually create a view for anonymous users to create a partner to submit for Bank approval.

File: src/Controller/ClientController.php [hunk 3]
```diff
@@ -145,6 +154,8 @@ public function update(Request $request, string $uuid): JsonResponse
      * Delete a client
      *
      * @Route(path="/{uuid}", methods={"DELETE"})
+     * @IsGranted({"ROLE_CLIENT_EDIT_ALL","ROLE_CLIENT_MANAGE_OWN"})
```
Comment (akoebbe) [inline src/Controller/ClientController.php:157]: We should probably just make this available only to admins. In theory, a client should never be deleted, it should only be changed to inactive.

File: src/Controller/OrderController.php [hunk 4]
```diff
@@ -126,9 +133,10 @@ public function update(Request $request, $id)
     }
 
     /**
-     * Delete a order
+     * Delete an order
      *
      * @Route(path="/{id<\d+>}", methods={"DELETE"})
+     * @IsGranted({"ROLE_ORDER_EDIT_ALL","ROLE_ORDER_MANAGE_OWN"})
```
Comment (akoebbe) [inline src/Controller/OrderController.php:139]: Let's make this only available for admins for right now.

File: src/Controller/OrderController.php [hunk 5]
```diff
@@ -152,7 +160,9 @@ public function destroy($id)
     }
 
     /**
+     *
      * @Route(path="/bulk-change", methods={"PATCH"})
+     * @IsGranted({"ROLE_ORDER_EDIT_ALL","ROLE_ORDER_MANAGE_OWN"})
```
Comment (akoebbe) [inline src/Controller/OrderController.php:165]: I don't think there will be a need for partners to bulk change their orders. Let's pull off `ROLE_ORDER_MANAGE_OWN`

File: src/Controller/OrderController.php [hunk 6]
```diff
@@ -180,7 +190,9 @@ public function bulkChange(Request $request)
     }
 
     /**
+     *
      * @Route(path="/bulk-delete", methods={"PATCH"})
+     * @IsGranted({"ROLE_ORDER_EDIT_ALL","ROLE_ORDER_MANAGE_OWN"})
```
Comment (akoebbe) [inline src/Controller/OrderController.php:195]: Let's make this one only for admins.

File: src/Controller/PartnerController.php [hunk 7]
```diff
@@ -146,7 +169,10 @@ public function clients(Request $request, int $id): JsonResponse
     }
 
     /**
+     *
      * @Route("/{id<\d+>}/transition", methods={"PATCH"})
+     * @IsGranted({"ROLE_PARTNER_EDIT_ALL","ROLE_PARTNER_MANAGE_OWN"})
```
Comment (akoebbe) [inline src/Controller/PartnerController.php:174]: Only admins and managers should be able to transition statuses on partners. Let's pull `ROLE_PARTNER_MANAGE_OWN` off.

**Comment by uplyft** [comment]
Thanks @akoebbe! Let me know what you think of the latest?

**Pull request merged**